### PR TITLE
RDKB-52189 : Migrate MQTTCM to use Xpki operational cert from secure element

### PR DESCRIPTION
**Pull request by nlrcomcast**

### CODE REVIEW

File: src/mqttcm_connect.c [hunk 1]
```diff
@@ -414,6 +416,49 @@ int validateForMqttInit()
...
+
+	get_from_file("private_key_passwd=", &passphrase, "/dev/stdin");
+
+	if( !passphrase )
+	{
+		MqttCMError("Passkey not Found\n");
+	}
+	else
+	{
+		strncpy(buf, passphrase, size);
```
Comment (sadhyama) [inline src/mqttcm_connect.c:454]: where buf is getting allocated ? buf NULL check is needed .
Also, add the unit test for buf Null case

File: tests/test_mqttcm_connect.c [hunk 2]
```diff
@@ -36,7 +36,7 @@ extern pthread_cond_t mqtt_retry_con;
 extern pthread_mutex_t mqtt_mut;
 extern pthread_cond_t mqtt_con;
 extern rbusHandle_t rbus_handle;
-
+int password_callback(char *buf, int size, int rwflag, void *userdata);
```
Comment (sadhyama) [inline tests/test_mqttcm_connect.c:39]: Add password_callback declaration in ../src/mqttcm_connect.h and avoid defining again here.

File: src/mqttcm_connect.c [hunk 3]
```diff
@@ -414,6 +416,49 @@ int validateForMqttInit()
 	return 0;
 }
 
+void custom_log_callback(struct mosquitto *mosq, void *userdata, int level, const char *message)
+{
+    // Control logging based on levels
+    if (level == MOSQ_LOG_DEBUG) {
+        // Log debug messages
```
Comment (sadhyama) [inline src/mqttcm_connect.c:423]: Remove this additional comments and keep only "// Control logging based on levels"

File: tests/test_mqttcm_connect.c (hunk 2 above)
Comment (sadhyama) [reply tests/test_mqttcm_connect.c:39]: Same for custom_log_callback also

File: src/mqttcm_connect.c [hunk 4]
```diff
@@ -180,8 +182,6 @@ bool mqttCMConnectBroker()
 			username = clientId;
 			MqttCMInfo("clientId is %s username is %s\n", clientId, username);
 
-			execute_mqtt_script(OPENSYNC_CERT);
```
Comment (sadhyama) [inline src/mqttcm_connect.c:183]: Do we need this macro anymore OPENSYNC_CERT? If not remove it from .h

File: tests/test_mqttcm_webcfg.c [hunk 5]
```diff
@@ -154,6 +154,7 @@ void test_sendRbusEventWebcfgOnSubscribe_success()
 	rbusError_t ret = sendRbusEventWebcfgOnSubscribe();
 	CU_ASSERT_EQUAL(ret, 0);
 	rbushandleclose(WEBCFG_MQTT_SUBSCRIBE_CALLBACK);
+	sleep(1);//It is taking some time to unsubscribe above event, so adding delay
```
Comment (sadhyama) [inline tests/test_mqttcm_webcfg.c:157]: Add rbus is taking time to unscribe the event . same for other places

File: src/mqttcm_connect.c [hunk 6]
```diff
@@ -414,6 +416,49 @@ int validateForMqttInit()
 	return 0;
 }
 
+void custom_log_callback(struct mosquitto *mosq, void *userdata, int level, const char *message)
```
[the remark this answers is not in the document]
Comment (nlrcomcast) [reply src/mqttcm_connect.c:417]: Added

File: src/mqttcm_connect.c (hunk 1 above)
Comment (nlrcomcast) [reply src/mqttcm_connect.c:458]: Added size and buf check, added unit test cases for null check and size

File: tests/test_mqttcm_connect.c (hunk 2 above)
Comment (nlrcomcast) [reply tests/test_mqttcm_connect.c]: Removed in c file and added in header file

File: src/mqttcm_connect.c (hunk 3 above)
Comment (nlrcomcast) [reply src/mqttcm_connect.c]: Removed

File: src/mqttcm_connect.c (hunk 4 above)
Comment (nlrcomcast) [reply src/mqttcm_connect.c:183]: not required, removed

File: tests/test_mqttcm_webcfg.c (hunk 5 above)
Comment (nlrcomcast) [reply tests/test_mqttcm_webcfg.c]: updated comment
